core/domain/state_domain.py

```diff
@@ -184,13 +184,18 @@ def get_all_html_content_strings(self):
                         interaction_and_rule_details['ruleTypes'][
                             rule_spec.rule_type][
                                 'htmlInputVariables'])
+                    if len(rule_spec.inputs.keys()) == 0:
+                        raise Exception(
+                            'Rule spec should have atleast one input variable.')
```

Comment:
atleast --> at least (I feel like I've commented on this sort of thing before, btw)

Also this isn't the check I was expecting -- rules can legitimately have zero input variables. I meant that, in your for loop below, you have a top-level "if" check, and you're expecting that "if" check to trigger at least once, right? Check that it does actually trigger at least once.

Reply:
done, sorry for misinterpreting.